Server/source/Utils.cpp

```
@@ -1,38 +1,25 @@
 #include "Utils.h"
-
-using namespace std;
+#include <cstring>
 
 namespace Utils
 {
-NsApplicationControlData *getAppControlData(u64 tid)
+const char *getAppName(u64 application_id)
 {
-    NsApplicationControlData *appControlData = new NsApplicationControlData();
+    static NsApplicationControlData appControlData;
     size_t appControlDataSize = 0;
+    NacpLanguageEntry *languageEntry = nullptr;
 
-    Result rc;
-    rc = nsGetApplicationControlData(NsApplicationControlSource_Storage, tid, appControlData, sizeof(NsApplicationControlData), &appControlDataSize);
-    if (R_FAILED(rc))
-    {
-        delete appControlData;
-        fatalThrow(rc);
-    }
+    memset(&appControlData, 0x00, sizeof(NsApplicationControlData));
 
-    return appControlData;
-}
+    if (R_FAILED(nsGetApplicationControlData(NsApplicationControlSource_Storage, application_id, &appControlData, sizeof(NsApplicationControlData), &appControlDataSize)))
+        return "A Game";
```

It's clear that all of these "A Game" returns are failure conditions. You can either re-conceptualize your branching statements (like if R_SUCCEEDED instead of R_FAILED) so that returning "A Game" is the last thing you do, or make a failure label and jump to it.